src/qutip_qip/operations/gates.py

```diff
@@ -57,7 +57,7 @@
            'gate_expand_1toN', 'gate_expand_2toN', 'gate_expand_3toN',
            'qubit_clifford_group', 'expand_operator', '_single_qubit_gates',
            '_para_gates', '_ctrl_gates','_swap_like','_toffoli_like',
-           '_fredkin_like']
+           '_fredkin_like','_GATE_NAME_TO_LABEL','_gate_label']
```

Comment:
We shouldn't add these `_` constants and functions to `__all__` I don't think and we don't need to import `_GATE_NAME_TO_LABEL` in `circuit.py` at all.

Reply:
> we don't need to import _GATE_NAME_TO_LABEL in circuit.py at all

We do actually. It's used in `circuit.py`. I tried running one of the notebooks with `_gate_label` and that caused some errors.

Reply:
```
$ ag _GATE_NAME_TO_LABEL .
src/qutip_qip/operations/gates.py
171:            if (name in _GATE_NAME_TO_LABEL) and (arg_value is not None):
226:_GATE_NAME_TO_LABEL = {
263:    if name in _GATE_NAME_TO_LABEL:
264:        gate_label = _GATE_NAME_TO_LABEL[name]
```
I'm not seeing it used anywhere other than in `gates.py`? Could you provide a link to where else it is used?

Reply:
Sorry, I was confused between `_gate_label` and `_GATE_NAME_TO_LABEL`. You are right. Only `_gate_label` has to be imported. 

Although, for this issue[ Boxi wanted a separate PR](https://github.com/qutip/qutip-qip/issues/60#issuecomment-864374124). Feel free to close this PR if that's changed.

Reply:
Oh, I was asking for a separate PR mainly because the decomposition is a new feature and this is a bug fix. Mixing them might cause some confusion if someone checks the history one day. As #66 is all about bug fixing and cleaning, I can just merge that one and saves you the trouble of updating it for `_GATE_NAME_TO_LABEL`. But thank you anyway!